rep-0150.rst

```diff
@@ -0,0 +1,208 @@
+REP: 150
+Title: ROS Melodic and Newer Metapackages
+Author: Tully Foote <tfoote@openrobotics.org>, Chris Lalancette <clalancette@openrobotics.org>
+Status: Active
+Type: Informational
+Content-Type: text/x-rst
+Created: 03-Nov-2017
+Post-History:
+
+
+Abstract
+========
+
+This REP describes the metapackages for the ROS Melodic release and for releases that follow until otherwise replaced by a similar update REP.
+This REP updates REP 131 [3]_ and REP 142 [5]_ for changes in packages.
+
+
+Motivation
+==========
+
+For a discussion on the general motivation and role of variants,
+please see REP 108 [1]_.
+
+As of ROS Hydro we have completed the transition to using
+metapackages instead of variants.
+For more information on metapackages see REP 127 [2]_.
+
+This document covers the common metapackages used for recommended
+installations.
+It is expressed in the ros/metapackages repository on github.com
+[4]_.
+
+This REP proposes the same entrypoints as REP 108 and merely updates
+the variant definitions to reflect the organizational changes in ROS
+stacks.
+In particular, in ROS Melodic the `robot` metapackage is contracted to remove
+`robot_model` and instead explicitly list the necessary packages.
+
+The definitions in this REP are more focused and concise for clearer
+purpose, and do not list all transitive dependencies explicitly.
+
+
+Specification
+=============
+
+End-user entry points
+---------------------
+
+We define the same three main entry points for ROS users as REP 108
+[1]_.
+
+ * desktop_full (recommended)
+ * desktop
+ * ros_base
+
+Metapackages
+------------
+
+ROS Core
+''''''''
+
+The `ros_core` metapackage composes the core communication protocols.
+It may not contain any GUI dependencies.
+
+::
+
+ - ros_core:
+      extends: [ros, ros_comm]
+      packages: [catkin, cmake_modules, common_msgs, console_bridge,
+                 gencpp, geneus, gennodejs, genlisp, genmsg, genpy,
+                 message_generation, message_runtime, rosbag_migration_rule,
+                 rosconsole_bridge, roscpp_core, rosgraph_msgs, roslisp,
+                 rospack, std_msgs, std_srvs]
+
+ROS Base
+''''''''
+
+The `ros_base` metapackage composes the `ros_core` metapackage with
+commonly used libraries.
+It may not contain any GUI dependencies.
+
+::
+
+  - ros_base:
+      extends: [ros_core]
+      packages: [actionlib, angles, bond_core, class_loader,
```

Comment:
Removed.

Reply:
Not sure if the it's REP or the metapackage that should be modified. Maybe @tfoote or @dirk-thomas have some feedback?

Reply:
I'd keep angles out of the ros_base. It's certainly low level, but it's really a library that is a build and exec dependency with no user entry points so it will be brought in by any dependencies necessary but isn't actually useful on its own.

Reply:
`angles` should not just be removed. It needs to be moved to the variant which it is actually included in instead. Since it is a dependency of `turtle_actionlib` which is part of `common_tutorials` it should go into the same variant.